esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,381 @@
+             :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            default_privileges
+        )
+
+        for p in privileges:
+            p['tenant_id'] = id
+        self.conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+             :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            privileges
+        )
+        self.conn.commit()
+        return tenant
+    
+    def list_tenants(self):
+        """
+            Return a list of DockerVolumeTenants objects.
+        """
+        tenant_list = []
+        cur = self.conn.execute(
+           "select * from tenants"
+        )
+        result = cur.fetchall()
+       
+        for r in result:
+            # loop through each tenant
+            id = r['id']
+            name = r['name']
+            description = r['description']
+            default_datastore = r['default_datastore']
+            
+            # search vms for this tenant
+            vms = []
+            cur = self.conn.execute(
+                "select * from vms where tenant_id=?",
+                (id,)
+            )
+            vms = cur.fetchall()
+            
+            # search privileges and default_privileges for this tenant
+            privileges = []
+            cur = self.conn.execute(
+                "select * from privileges where tenant_id=? and datastore!=?",
+                (id,default_datastore)    
+            )
+            privileges = cur.fetchall()
+
+            default_privileges = []
+            cur = self.conn.execute(
+                "select * from privileges where tenant_id=? and datastore=?",
+                (id,default_datastore)    
+            )
+            default_privileges = cur.fetchall()
+            tenant = DockerVolumeTenant(name, description, default_datastore,
+                                        default_privileges, vms, privileges, id)
+            tenant_list.append(tenant)
+
+        return tenant_list   
+    
+    def remove_volumes_from_volume_table(self, tenant_id):
+          
+        self.conn.execute(
+                "delete from volumes where tenant_id=?", 
+                [tenant_id]
+        )
+
+        self.conn.commit()
+    
+    def remove_volumes_for_tenant(self, tenant_id):
+        cur = self.conn.execute(
+           "select * from tenants where id=?",
+           (tenant_id,)
+        )
+        result = cur.fetchone()
+        logging.debug("remove_volumes_for_tenant: %s %s", tenant_id, result)
+        tenant_name = result['name']
+        vmdks = vmdk_utils.get_volumes(tenant_name)
+        # Delete all volumes for this tenant. 
+        # Do we need to remove the path /vmfs/volumes/datastore_name/tenant_name??
+        for vmdk in vmdks:
+            vmdk_path = vmdk['path']+"/"+vmdk['filename']
```

Comment:
Please do not forget to add spaces around ops like '+'.

Also, I think we have something in vmdk_utils for building ful vmdk filename - please use it.
If we don't, then add it (and use os.path.join insidem instead of string '+')